bounties_api/std_bounties/management/commands/bounties_subscriber.py

```diff
@@ -37,121 +164,20 @@ def handle(self, *args, **options):
+                logger.info('done processing regular message, retry_blacklist is now: {}'.format(retry_blacklist))
 
-                if event == 'ContributionAdded':
-                    master_client.contribution_added(bounty_id,
-                                                     event_date=event_date,
-                                                     inputs=contract_method_inputs,
-                                                     event_timestamp=event_timestamp)
-
-                if event == 'DeadlineExtended':
-                    master_client.deadline_extended(bounty_id,
-                                                    event_date=event_date,
-                                                    inputs=contract_method_inputs,
-                                                    event_timestamp=event_timestamp)
-
-                if event == 'BountyChanged':
-                    master_client.bounty_changed(bounty_id,
-                                                 event_date=event_date,
-                                                 inputs=contract_method_inputs)
-
-                if event == 'IssuerTransferred':
-                    master_client.issuer_transferred(bounty_id,
-                                                    transaction_from=transaction_from,
-                                                    event_date=event_date,
-                                                    inputs=contract_method_inputs)
-
-                if event == 'PayoutIncreased':
-                    master_client.payout_increased(bounty_id,
-                                                   event_date=event_date,
-                                                   inputs=contract_method_inputs)
-
-                logger.info(event)
-
-                # This means the contract subscriber will never send this event
-                # through to sqs again
-
-                Event.objects.get_or_create(
-                    event=event,
-                    transaction_hash=transaction_hash,
-                    defaults = {
-                        'bounty_id': bounty_id,
-                        'fulfillment_id': fulfillment_id if fulfillment_id != -1 else None,
-                        'transaction_from': transaction_from,
-                        'contract_inputs': contract_method_inputs,
-                        'event_date': event_date,
-                    }
-                )
-                redis_client.set(message_deduplication_id, True)
-                sqs_client.delete_message(
-                    QueueUrl=settings.QUEUE_URL,
-                    ReceiptHandle=receipt_handle,
-                )
+                if retry_blacklist <= 0:
+                    retry_blacklist = retry_blacklist_rate
+                    logger.info('retry_blacklist reset to: {}'.format(retry_blacklist))
+                    retry = redis_client.lrange('retry_blacklist', 0, -1)
+                    for blacklisted in retry:
```

Comment:
definitely moving in the right direction... a couple issues:
1. We're storing a bounty id in the blacklisted, not a message - so this wouldn't function correctly.
2. Re constructing the message format is heavy... it's probably easier to have a separate function that unwraps the message... this way we can still share a function for blacklisted and messages.
3. We would need to query the Event object here to get the inputs... Keep in mind, an event can have multiple transactions on it. Even a transaction hash can have multiple events on it. This is why the dedupe id is a combination of the transaction hash and the event name.
4. We need a way when querying the Event model to get the events that have not been written yet, and in order :). As events related to a bounty cannot be processed out of order. The best way here is ot query by date, then you have to check redis at `redis_client.get(message_deduplication_id)` to see if it has been handled yet.

Reply:
@codeluggage if you want to keep this format... of handling the message. An easier solution is to just raise the error and throw the try/catch in this main block of code then add the message directly to the blacklisted array.  This whole block of code can check for the StatusError from ipfs.  Thoughts? I think that would DRASTICALLY simplifiy things.

Reply:
And a much cleaner solution come to think of it. This would require minimal changes from this current code :)

Reply:
But would still need to block future transactions that come through for the same bounty_id.